ext/date/php_date.c

```diff
@@ -5279,7 +5279,14 @@ static zval *date_period_read_property(zval *object, zval *member, int type, voi
 /* {{{ date_period_write_property */
 static void date_period_write_property(zval *object, zval *member, zval *value, void **cache_slot)
 {
-	zend_throw_error(NULL, "Writing to DatePeriod properties is unsupported");
+	if (!std_object_handlers.has_property(object, member, 2, NULL)) {
+		zend_string *name = zval_get_string(member);
+		zend_throw_error(NULL, "Writing to DatePeriod->%s is unsupported", ZSTR_VAL(name));
+		zend_string_release(name);
```

Comment:
I added this line because Travis was reporting a memory leak, but even after adding this it's still leaking.
I've looked at the lines reported, and I can see memory being allocated, but I can't see how it's related to these changes, can anybody provide any help?

```
[Tue Feb 20 22:04:54 2018]  Script:  '/tmp/php-src/ext/date/tests/DatePeriod_properties1.php'
/tmp/php-src/ext/date/php_date.c(2164) :  Freeing 0x00007fa952e5ac60 (56 bytes), script=/tmp/php-src/ext/date/tests/DatePeriod_properties1.php
/tmp/php-src/Zend/zend_alloc.c(2488) : Actual location (location was relayed)
Last leak repeated 1 time
[Tue Feb 20 22:04:54 2018]  Script:  '/tmp/php-src/ext/date/tests/DatePeriod_properties1.php'
/tmp/php-src/ext/date/php_date.c(2425) :  Freeing 0x00007fa952e5acc0 (64 bytes), script=/tmp/php-src/ext/date/tests/DatePeriod_properties1.php
/tmp/php-src/Zend/zend_alloc.c(2488) : Actual location (location was relayed)
=== Total 3 memory leaks detected ===
```

Reply:
It seems to me that the memory error is not necessarily related to your changes, but rather that DatePeriod_properties1.phpt exhibits an already existing memory leak.